src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -680,32 +682,117 @@ public bool Equals([NotNullWhen(true)] string? value, StringComparison compariso
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ulong ReadUInt64(string str) =>
+            Unsafe.ReadUnaligned<ulong>(ref Unsafe.As<char, byte>(ref str._firstChar));
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ulong ReadUInt64(string str, nint offset) =>
+            Unsafe.ReadUnaligned<ulong>(ref Unsafe.As<char, byte>(ref Unsafe.Add(ref str._firstChar, offset)));
+
         // Determines whether two Strings match.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool Equals(string? a, string? b)
         {
-            // Transform 'str == ""' to 'str != null && str.Length == 0' if either a or b are jit-time
-            // constants. Otherwise, these two blocks are eliminated
-            if (RuntimeHelpers.IsKnownConstant(a) && a != null && a.Length == 0)
+#if TARGET_64BIT && !MONO && !BIGENDIAN
+            // Try to unroll Equals in case of constant 'b' for b.Length in [0..16] range
+            if (RuntimeHelpers.IsKnownConstant(b) && !RuntimeHelpers.IsKnownConstant(a) && b != null)
             {
-                return b != null && b.Length == 0;
+                // Unroll using SWAR
+                if (b.Length <= 8)
+                {
+                    return EqualsUnrolled_0_to_8(a, b);
+                }
+                // Unroll using two Vector128s
+                if (b.Length >= 9 && b.Length <= 16 && Vector128.IsHardwareAccelerated)
```

Comment:
```suggestion
                if (b.Length <= 16 && Vector128.IsHardwareAccelerated)
```
Interval `[0, 8]` already covered by previous check, so only `(8, b.Length]` remains, thus `b.Length >= 9` is redundant.